test/unit/math/prim/fun/zeros_array_test.cpp

```diff
@@ -0,0 +1,23 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+#include <vector>
+
+TEST(MathFunctions, zeros_array) {
+  using stan::math::zeros_array;
+
+  std::vector<double> u0 = zeros_array(0);
+  EXPECT_EQ(0, u0.size());
+
+  for (int K = 1; K < 5; K++) {
+    std::vector<double> u = zeros_array(K);
+    std::vector<double> v(K);
```

Comment:
Same here as below---`v` needs to be initialized to 0 and the 0 case can be rolled into the loop.